src/EnergyPlus/BoilerSteam.cc

```diff
@@ -98,11 +98,11 @@ namespace BoilerSteam {
         }
 
         // Now look for this particular pipe in the list
-        for (auto &boiler : state.dataBoilerSteam->Boiler) {
-            if (boiler.Name == objectName) {
-                return &boiler;
-            }
-        }
+        auto myBoiler = std::find_if(state.dataBoilerSteam->Boiler.begin(),
+                                     state.dataBoilerSteam->Boiler.end(),
+                                     [&objectName](const BoilerSpecs &boiler) { return boiler.Name == objectName; });
+        if (myBoiler->Name == objectName) return myBoiler;
+
         // If we didn't find it, fatal
         ShowFatalError(state, format("LocalBoilerSteamFactory: Error getting inputs for steam boiler named: {}", objectName)); // LCOV_EXCL_LINE
         // Shut up the compiler
```

Comment:
Here's most of the list for this branch:

    [src/EnergyPlus/BoilerSteam.cc:102]:(style),[useStlAlgorithm],Consider using std::find_if algorithm instead of a raw loop.
    [src/EnergyPlus/Boilers.cc:105]:(style),[useStlAlgorithm],Consider using std::find_if algorithm instead of a raw loop.
    [src/EnergyPlus/CTElectricGenerator.cc:110]:(style),[useStlAlgorithm],Consider using std::find_if algorithm instead of a raw loop.
    [src/EnergyPlus/ChillerAbsorption.cc:121]:(style),[useStlAlgorithm],Consider using std::find_if algorithm instead of a raw loop.
    [src/EnergyPlus/ChillerElectricASHRAE205.cc:462]:(style),[useStlAlgorithm],Consider using std::find_if algorithm instead of a raw loop.
    [src/EnergyPlus/ChillerElectricEIR.cc:124]:(style),[useStlAlgorithm],Consider using std::find_if algorithm instead of a raw loop.
    [src/EnergyPlus/ChillerExhaustAbsorption.cc:122]:(style),[useStlAlgorithm],Consider using std::find_if algorithm instead of a raw loop.
    [src/EnergyPlus/ChillerGasAbsorption.cc:289]:(style),[variableScope],The scope of the variable 'ChillerName' can be reduced.
    [src/EnergyPlus/ChillerGasAbsorption.cc:119]:(style),[useStlAlgorithm],Consider using std::find_if algorithm instead of a raw loop.
    [src/EnergyPlus/ChillerIndirectAbsorption.cc:120]:(style),[useStlAlgorithm],Consider using std::find_if algorithm instead of a raw loop.
    [src/EnergyPlus/ChillerReformulatedEIR.cc:125]:(style),[useStlAlgorithm],Consider using std::find_if algorithm instead of a raw loop.
    [src/EnergyPlus/Coils/CoilCoolingDXCurveFitPerformance.cc:448]:(style),[variableScope],The scope of the variable 'RedCapNum' can be reduced.